src/psyclone/dynamo0p3.py

```diff
@@ -9151,17 +8029,27 @@ def intrinsic_type(self):
                   or of the argument's data for fields and operators.
         :rtype: str
 
+        :raises InternalError: for an unsupported metadata for descriptor \
+                               data type.
+
         '''
+
+        try:
+            self._intrinsic_type = MAPPING_DATA_TYPES[
```

Comment:
I don't think we should be setting `_intrinsic_type` here. Could we do it in the constructor or just do away with it altogether?

Reply:
Moved to the `DynKernelArgument` constructor together with the `InternalError` check and description. Adjusted the `test_arg_intrinsic_type_error` in `dynamo0p3_test.py` accordingly.